### DISCUSSION TITLE
Encoding problems

### DISCUSSION THREAD

**Discussion by KLASTER911** [schemathesis/schemathesis#1126] [Q&A] [answered]
If the file imported via 'from_path' contains the Russian letter 'И', then there is a problem with the encoding.
Error: UnicodeDecodeError: 'charmap' codec can't decode byte 0x98 in position X: character maps to <undefined>
Can this problem be solved?
I was able to solve it when I added the encoding = 'utf-8' parameter to the original file 'loaders'.
![image](https://user-images.githubusercontent.com/18378562/114582745-8faf1200-9c89-11eb-9eb1-44daf32b458d.png)

**Comment by Stranger6667** [comment, marked as answer]
Hi!

Does it happen on Windows? If so, then it is related to 50999b9c0d6b2ea05385fcfd3f9fc81d6229ad19 which I didn't apply to other places

**Reply by KLASTER911** [reply]
Yes, it's on Windows. When you open an json or yaml file through a cmd or python script.

**Reply by Stranger6667** [reply]
Thanks! I will fix it soon

**Reply by Stranger6667** [reply]
I can't reproduce it. What version of Schemathesis & Python do you use?

**Reply by KLASTER911** [reply]
Python 3.9.2, 
schemathesis version 3.6.1

**Reply by Stranger6667** [reply]
Hm, I still can't reproduce :( Could you, please, provide some code sample that fails on your machine? And also the output of:

```python
import locale
import sys

print(locale.getpreferredencoding())
print(sys.flags.utf8_mode)
print(sys.getdefaultencoding())
```

In the end, if I won't be able to reproduce it, I am ok with just adding `encoding="utf-8"` wherever it is needed if it won't break existing code.

**Reply by KLASTER911** [reply]
![image](https://user-images.githubusercontent.com/18378562/114675450-e90e5400-9d10-11eb-8746-a765f11afb58.png)
![image](https://user-images.githubusercontent.com/18378562/114676373-e7915b80-9d11-11eb-97c8-209dbd22ece5.png)

my json file encoding is utf-8
I use windows 10 pro and system language is russian

**Reply by Stranger6667** [reply]
Could you please provide code as text and not as an image? So I can copy paste it

**Reply by KLASTER911** [reply]
```
import schemathesis
from hypothesis import settings
schema = schemathesis.from_path('C:\\Users\\Admin\\Desktop\\openapinew.json', base_url="http://127.0.0.1:8081")

def test_api(case):
    case.call_and_validate()
```

**Reply by Stranger6667** [reply]
FYI - the fix was released in `3.6.2`

**Comment by KLASTER911** [comment]
Thank you!
